src/rrs/pprint.3

```diff
@@ -295,7 +295,30 @@ PP	;"OBLIST"
 	      (<PRINC !" > <.COMELE <REST .L>>)> >>
 
 
+
 <DEFINE FORMAHEAD (ML "AUX" (AVSP <- <LINLNT .OUTCHAN> <LINPOS .OUTCHAN> .M>))
+	<COND (<AND <==? <TYPE <1 .ML>> FORM>
+		    <NOT <EMPTY? <REST .ML>>>
+		    <NOT <FLATSIZE <1 .ML> <MIN 59 .AVSP>>>>
+	       T)
+	      (ELSE
+	       <REPEAT ()
+		  ;"<COND (<L? <LENGTH .ML> 2> <RETURN #FALSE ()>) WAS"
+ 		  <COND (<L=? <LENGTH .ML> 2> <RETURN #FALSE ()>)
+			(<NOT <==? <TYPE <2 .ML>> FORM>> <RETURN #FALSE ()>)
+			(<EMPTY? <2 .ML>> <RETURN #FALSE()>)
+			(<FLATSIZE <1 <2 .ML>>
+				   <- <SET AVSP
+					   <- .AVSP
+					      3
+					      <FLATSIZE <1 .ML> 99999999>>>
+				      3>>
+			 <SET ML <2 .ML>>)
+			(ELSE <RETURN T>)>>)>>
+
+
+
+;<DEFINE FORMAHEAD (ML "AUX" (AVSP <- <LINLNT .OUTCHAN> <LINPOS .OUTCHAN> .M>))
```

Comment:
The old FORMAHEAD seems to be left in place, commented out.

Reply:
Yes, that's true. These changes correct FORMAHEAD hanlding of an empty FORMs

In FORMAHEAD to produce better looking indentation in PPRINT's output,
306:		  ;"<COND (<L? <LENGTH .ML> 2> <RETURN #FALSE ()>) WAS"
307: 		  <COND (<L=? <LENGTH .ML> 2> <RETURN #FALSE ()>)
308:			(<NOT <==? <TYPE <2 .ML>> FORM>> <RETURN #FALSE ()>)

Changes to:
306:		  <COND (<L? <LENGTH .ML> 2> <RETURN #FALSE ()>)
307: 		  ;<COND (<L=? <LENGTH .ML> 2> <RETURN #FALSE ()>)
308:			(<NOT <==? <TYPE <2 .ML>> FORM>> <RETURN #FALSE ()>)

RRS